tests/unit_tests/dft/include/compute_inplace.hpp

```diff
@@ -23,101 +23,70 @@
-                output_ref[j * last_dim_size + i / 2].imag();
-        }
-    }
-    return conjugate_even_ref;
-}
-
-template <typename T, typename al>
-void copy_strided(const std::vector<std::int64_t>& sizes, const std::vector<T>& input,
-                  std::vector<T, al>& output) {
-    auto in_iter = input.cbegin();
-    auto out_iter = output.begin();
-    const auto row_len = static_cast<std::ptrdiff_t>(sizes.back());
-    const auto conjugate_row_len = static_cast<std::ptrdiff_t>(
-        row_elements_to_conjugate_even_components(cast_unsigned(row_len)));
-    while (in_iter < input.cend()) {
-        std::copy(in_iter, in_iter + row_len, out_iter);
-        in_iter += row_len;
-        out_iter += conjugate_row_len;
-    }
-}
-
 template <oneapi::mkl::dft::precision precision, oneapi::mkl::dft::domain domain>
 int DFT_Test<precision, domain>::test_in_place_buffer() {
     if (!init(MemoryAccessModel::buffer)) {
         return test_skipped;
     }
 
-    const std::size_t last_dim_size = cast_unsigned(sizes.back());
-    const std::size_t real_first_dims = size_total / last_dim_size;
-    const std::size_t real_last_dim = row_elements_to_conjugate_even_components(last_dim_size);
-    const std::size_t container_size_total =
-        domain == oneapi::mkl::dft::domain::REAL ? real_first_dims * real_last_dim : size_total;
-    const std::int64_t container_size_per_transform =
-        static_cast<std::int64_t>(container_size_total) / batches;
-    const std::int64_t backward_elements = domain == oneapi::mkl::dft::domain::REAL
-                                               ? container_size_per_transform / 2
-                                               : container_size_per_transform;
+    auto strides_fwd = this->strides_fwd;
+    auto strides_bwd = this->strides_bwd;
+    if (domain == oneapi::mkl::dft::domain::REAL) {
+        // both input and output strides must be set
+        if (strides_fwd.size() == 0) {
+            auto strides_tmp = get_conjugate_even_complex_strides(sizes);
+            strides_fwd = { strides_tmp[0] };
+            //to be able to calculate in place each row must fit backward data
+            for (size_t i = 0; i < sizes.size() - 1; i++) {
+                strides_fwd.push_back(strides_tmp[i + 1] * 2);
+            }
+            strides_fwd.push_back(1);
+        }
+        if (strides_bwd.size() == 0) {
+            auto strides_tmp = get_conjugate_even_complex_strides(sizes);
+            strides_bwd = { strides_tmp[0] };
+            //to be able to calculate in place each row must fit backward data
+            for (size_t i = 0; i < sizes.size() - 1; i++) {
+                strides_bwd.push_back(strides_tmp[i + 1]);
+            }
+            strides_bwd.push_back(1);
+        }
+    }
+    else {
+        // spec says strides_bwd is ignored and strides_fwd is reused for backward domain for in-place complex
+        strides_fwd = strides_bwd;
```

Comment:
It works either as it is or reversed. Just both must have the same value. Since we are testing transforms in both directions it also at some point becomes INPUT_STRIDES.